app/core/database.py

```
@@ -0,0 +1,113 @@
+class DatabaseManager:
+    def __init__(self):
+        """
+        Initialize a DatabaseManager with all client and database slots unset.
+
+        Attributes:
+            mongodb_client (Optional[AsyncIOMotorClient]): MongoDB async client instance or None until connected.
+            redis_client (Optional[redis.Redis]): Redis client instance or None until connected.
+            db: MongoDB database handle or None until a database is selected after connection.
+        """
+        self.mongodb_client: Optional[AsyncIOMotorClient] = None
+        self.redis_client: Optional[redis.Redis] = None
```

Ruff (0.13.3)

Undefined name `redis`

(F821)